### ISSUE TITLE
Unavailability

### ISSUE THREAD

**Issue by werebus** [umts/jobapps#132]
As Don was walking us through his interview process, one of the things that jumped out at me was this:

Folks come in for an interview and one of the first things they do is fill out a paper form that has their (un)availability to train on it.  If they're too busy, they get sent away at that point.

Would it be worth writing something to collect this information when they apply?  If we're looking to make it generic, we might not want to hard-code the time block definitions or requirements. However an (optional, depending on the position) unavailability gathering interface might be interesting.

**Comment by dfaulken** [comment]
It might, and seems like a generally useful tool. One wonders if a question of this nature could equally be added to the application.

**Comment by Anbranin** [comment]
Yessss.

**Comment by sherson** [comment]
Obviously solutions that don't require additional coding are appealing. Perhaps Don would be OK with a new section with seven text questions, each asking for day-of-week unavailability.

If that doesn't suffice, then a day-of-week unavailability grid ought to be generic enough to be useful for other purposes too.

**Comment by werebus** [comment]
I feel pretty confident in guessing that "Are you available any time on Tuesday?" isn't going to cut it. And also that questions like "Are you available 7:00am - 11:00am on Wednesday?" would be a little cumbersome.  One could do something like: (explain blocks in an explanation) "How many full blocks of time are you available for on weekdays?", but I bet it'll be more confusing than what they're doing now.

Basically, it has to be as easy to fill out *and* interpret as the paper sheet in order for anyone to want to use it.

**Comment by sherson** [comment]
I agree it's unlikely text questions would work well here. 

Something something feature creep, but it's worth looking into options to address this.

**Comment by bgregg** [comment]
I have spoken with Don about this. He says that while he doesn't like to just send people away because of their unavailability, having them fill it out before they apply can save some time, as the paper form can be cumbersome. He also says that he likes to have the option to work with someone who has a lot of unavailability. I can start working on putting a calendar similar to the paper sheet (Don gave me a  copy) into the application.

**Comment by bgregg** [comment]
![screen shot 2016-06-08 at 3 26 10 pm](https://cloud.githubusercontent.com/assets/14984384/15907990/6aab14a6-2d8d-11e6-8c79-ad59a14876c7.png)

This is what I've got so far. Still trying to figure out how to make the data from the checkboxes persist, but I've made some good progress. Thoughts as far as styling goes? (or otherwise but styling is mostly what I am concerned about right now)

**Comment by sherson** [comment]
I suggested to @bgregg that a grid with so many checkboxes would be cumbersome if the user can't select ranges (via dragging mouse, <alt><click>, whatever), and that there may be a plugin out there that already handles this.

**Comment by sherson** [comment]
Also, I asked Don if hourly blocks would suffice (thinking it'd make the UI cleaner) and he said that's fine.
